extension/memorylimiterextension/config.go

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package memorylimiterextension // import "go.opentelemetry.io/collector/extension/memorylimiterextension"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/internal/memorylimiter"
+)
+
+// MemoryLimiterExtension is an Extension control memory usage of OpenTelemetry Collector.
+// Common usage is to check memory usage during connection and can reject or accept the connection based on the
+// configured memoryLimiter settings
+type MemoryLimiterExtension interface {
+	// CheckMemory check if memory usage is above soft limits and return error to refuse data if is above.
+	CheckMemory() error
```

Comment:
I think `MustRefuse() bool` would be a better method to expose by the extension. @open-telemetry/collector-approvers WDYT?

Reply:
Please change it to `MustRefuse` unless there are any objections

Reply:
changed